app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -262,6 +269,37 @@ def remove_disk_config_spec(vim_obj, vmcs, hardware, options)
     end
   end
 
+  def resize_disk_config_spec(vim_obj, vmcs, hardware, options)
+    raise "resize_disk_config_spec: disk filename is required." unless options[:disk_name]
+
+    controller_key, key = vim_obj.getDeviceKeysByBacking(options[:disk_name], hardware)
+    raise "resize_disk_config_spec: no virtual device associated with: #{options[:disk_name]}" unless key
+
+    device = vim_obj.getDeviceByBacking(options[:disk_name])
+    raise "resize_disk_config_spec: decrease size is not supported for: #{options[:disk_name]}" unless device.capacityInKB.to_i <= options[:disk_size_in_mb].to_i * 1024
+
+    add_device_config_spec(vmcs, VirtualDeviceConfigSpecOperation::Edit) do |vdcs|
+      vdcs.device = VimHash.new("VirtualDisk") do |dev|
+        dev.key           = key
+        dev.capacityInKB  = options[:disk_size_in_mb].to_i * 1024
+        dev.controllerKey = controller_key
```

Comment:
Do you need the rest of these?  I think just key is required plus whatever you want to change (ref: https://www.vmware.com/support/developer/vc-sdk/visdk25pubs/ReferenceGuide/vim.vm.device.VirtualDevice.html)

Reply:
I did some testing.
Removing *dev.connectable* --> No problem, removing this
Removing *dev.backing* --> Error: Incompatible device backing specified for device '0'., keeping this
Removing *dev.unitNumber* --> Error: Invalid configuration for device '0', keeping this
Removing *dev.controllerKey* --> Error: Device requires a controller., keeping this

Do you know a quicker way testing these option, currently I do a evm:stop / evm:start to test code changes in provider code, is there a quicker way (I tried killing some workers (generic, refesh or broker), they are restarted but I keep hitting the old code ...)

Reply:
Interesting, so yes you can run in one console `bin/rails s` to run the UI and `bin/rails c` in another and run `simulate_queue_worker` in the rails c terminal.  That way you can just ctrl-C the worker Sim and `reload!` the code when you change something.

Reply:
A cool. I will try this!